packages/babel-plugin-transform-classes/README.md

```diff
@@ -6,8 +6,8 @@
 
 When extending a native class (e.g., `class extends Array {}`), the super class
 needs to be wrapped. This is needed to workaround two problems:
-- Babel transpiles classes using `SuperClass.apply(/* ... */)`, but native
-  classes aren't callable and thus throw in this case.
+- When extending native classes the `this.constructor` must not be tampered,
+  since it is the only way that Babel has of obtaining the class constructor.
```

Comment:
This affects all classes, not just the native ones 🤔

Reply:
Oh dang, would it be a regression?

Reply:
A tradeoff. The ability to mess with `this.constructor` for the ability to extend native classes.

I think it's fine to do it as a breaking change in babel 7.

Erasing `this.constructor` in itself won't break subclassing, because of the `|| Derived` in the helper:

```js
class A {}
class B extends A {}
delete B.prototype.constructor
;(new B()) instanceof B // still true
class C extends B {}
delete C.prototype.constructor
;(new C()) instanceof C // should also be true
```

But defining `constructor` to be something else can summon Cthulhu.